docs/rfcs/confmap-provider-configuration.md

```diff
@@ -0,0 +1,184 @@
+environment variables, or may be remotely accessed over the network. In the
+process of obtaining configuration from a source, the user may wish to modify
+the behavior of how the source is obtained.
+
+For example, consider the case where the Collector obtains configuration over
+HTTP from an HTTP endpoint. A user may want to:
+
+1. Poll the HTTP endpoint for configuration at a configurable interval and
+   reload the Collector service if the configuration changes.
+2. Authenticate the request to get configuration by including a header in the
+   request.
+
+This would produce a set of options like the following:
+
+- `poll-interval`: Sets an interval for the Provider to check the HTTP endpoint
+  for changes. If the config has changed, the service will be reloaded. Defaults
+  to `0`, which disables polling.
+- `headers`: Specifies a semicolon-delimited set of headers that will be passed
+  to the config backend. Header keys and values should be separated by `=`.
+
+Configuration for Providers cannot be stored in the same place as the rest of
+the Collector's configuration since the Providers are the ones tasked with
+getting that configuration. Therefore, they need a separate configuration
+mechanism.
+
+## Current state
+
+No upstream Providers currently offer any configuration options. The exported
+interfaces are still able to change before the `confmap` module is declared
+stable, but avoiding breaking changes in the API would be preferable.
+
+## Desired state
+
+We would like a mechanism to configure Providers that meets the following
+requirements:
+
+1. Broadly applicable to all current upstream providers.
+2. Can be used as a suggestion for configuring custom providers.
+3. Is consistent between Providers.
+4. Is configurable per config URI.
+
+## Solutions
+
+Providers are invoked through passing `--config` flags to the Collector binary
+with a scheme and URI to obtain. A single instance of each Provider is then
+tasked with retrieving from config from all URIs passed to the Collector for the
+scheme the provider is registered to handle.
+
+We have the following places where it may make sense to hook into options for
+Providers:
+
+1. Parts of the URI we are requesting.
+2. Part of the string passed to `--config`.
+3. Separate flags to configure Providers per config URI.
+4. Use a separate config file that specifies config sources inside a map
+   structure.
+5. Extend the Collector's config schema to support specifying additional places
+   to obtain configuration
+
+### Configure options inside the URI
+
+[RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3), which
+specifies the format of a URI, specifies the different parts of a URI and
+suggests two places where we could pass options to Providers: queries and
+fragments.
+
+#### Queries
+
+Advantages:
+
+- Explicitly intended to  specify non-hierarchical data in a URI.
+- Often used for this purpose.
+- Fits into existing config URIs.
```

Comment:
Not sure how well it fits with the `env` provider, I would expect people to want to use Bash-like syntax or Powershell-like syntax there

Reply:
Good point, I added a note to make this clear.